Spartacus

dury10 opened the pull request:
<!--
Thank you for your interest in contributing to the VoronUsers repository, it is
highly appreciated!
-->

<!--
Describe the submission further using the template provided below. The more
details the better! Please don't forget to thoroughly read the mod requirements and rules
and make sure your mod submission adheres to these rules.
Doing so speeds up the review process and will increase your chances at acceptance.
-->

### Which mods are added by this PR?

### How was it tested?

### Any background context you want to provide?

### Screenshots (if appropriate)

### ⚠️⚠️**PLEASE READ AND COMPLETE THE CHECKLIST BELOW**⚠️⚠️

  * [x] I have read the rules available [here](https://github.com/VoronDesign/VoronUsers/wiki/Mod-Submission-Rules) and
        my mod adheres to these rules.
  * [x] This mod was created by myself and I agree to publish it under the repository
        [license](https://github.com/VoronDesign/VoronUsers/blob/master/LICENSE.md)

dury10 commented:
Here is a live stream with details how to asamble the 
toolhead https://www.youtube.com/live/S5xbkWCNOBw?si=QTv8fzixqjN_HTtS

and here is a video with the Toolhead in action
https://www.youtube.com/watch?v=hGzCLkhoCfY


Regarding the if was tested. 
this toolhead was on a heavy load/test in the last 6 months 
![image](https://github.com/VoronDesign/VoronUsers/assets/35086459/97375e08-c16f-42ae-8178-2e4176ef8edc)

![Screenshot 2024-05-09 165606](https://github.com/VoronDesign/VoronUsers/assets/35086459/1c7cf329-740b-4bac-a3eb-5435be2003ed)

dury10 commented:
> 1. Can you please explain how this is different than a stretched Mini-Afterburner with some covers?
This Toolhead it has better part cooling it uses 2x 4010 
It has 2 Neopixel LED lights for improved visibility and real-time print status updates
It has pasive cooling for the nema14 stepper motor 
A lot of covers as you said with built in cable management. 

> 2. The description sounds like marketing but does not actually describe what the mod is.  Please update.
The description it has a clear key features and it says that it is a Toolhead, an alternative for what it exist already.


> 3. The readme absolutely cannot have a PayPal link or any mention of donations to you.
I have removed the link.

MapleLeafMakers commented on printer_mods/dury10/Spartacus/STL/shuter.stl:
this part looks like what would be called a `shuttle` in a voron toolhead.  that might be a more appropriate filename.

MapleLeafMakers commented on printer_mods/dury10/Spartacus/STL/45_degrees_strain_rellief_cover.stl:
`relief` only has one `l` 4 stl files here have it misspelled.

MapleLeafMakers commented on printer_mods/dury10/Spartacus/README.MD:
```diff
@@ -0,0 +1,115 @@
+
+### Nozzle Options (Optional, depending on preference):
+
+- T Volcano or Crazy Tvolcano Heat Block for easier nozzle changes and protect the heatbrake when printing at speeds.
```
```suggestion
- T Volcano or Crazy Tvolcano Heat Block for easier nozzle changes and protect the heatbreak when printing at speeds.
```

should `Crazy Tvolcano` be `NF Crazy Volcano`? I can't find anything called a `Crazy Tvolcano`

MapleLeafMakers commented on printer_mods/dury10/Spartacus/README.MD:
```diff
@@ -0,0 +1,115 @@
+
+#### This versatile and feature-packed 3D printer tool head provides an affordable solution for high-quality, high-speed printing, enhanced by Neopixel LED lights for improved visibility and dynamic print status updates.
+
+#### Assembly of Spartacus requires minimal tools, including an 1.5mm drill bit, a drill, an angle grinder or hacksaw, and 1.5, 2 and 2.5mm Allen keys. With these tools, users can effortlessly assemble Spartacus and unleash its full potential.
```
hacksaw and angle grinder aren't common tools for printer modding, when you add the assembly instructions, please be sure to explain their usage.

dury10 replied:
I will do my best to make a detailed install with photos. for the V6 that I have suggested. I have also added Rapido UHF that doesn't need cutting or drilling. 
Thanks for your time to review. I was already uploading the changes and after that I saw your suggestions, thanks

dury10 commented on printer_mods/dury10/Spartacus/README.MD:
```diff
@@ -0,0 +1,115 @@
+
+
+
+## Spartacus: The Toolhead stands as a testament to lightweight engineering mastery, boasting a total weight of under 320 grams, with the potential to further reduce weight by implementing titanium screws. Every component, from the stepper motor to the thermistor, board cables, heater, LEDs, fans, heatsink, nozzle, and screws, has been meticulously crafted to maintain this featherlight profile wi ...
```
English is not my first language. and yes i have used chatgpt

MapleLeafMakers replied:
It might be better to write something in your native language and run it through a translation service.  What you have here is extremely long, but says almost nothing about the toolhead.

- how is it different from the stock designs?
- who is the target audience?
- What voron machines can it be installed on?
- what were the design goals?

Your description should answer these questions in the first paragraph.

refrain from making unsubstantiated claims such as:
 - unmatched in its class
 - crafted with precision and integrity
 - stands as a testament to innovation


```
Every component, from the stepper motor to the thermistor, board cables, heater, LEDs, fans, heatsink, nozzle, and screws, has been meticulously crafted to maintain this featherlight profile without compromising on performance or durability
```

Here you are claiming to have crafted the heatsink, screws, fans nozzle and PCB all specifically for this toolhead?  they're just off the shelf parts that everyone is using.

Remember, this is *documentation*, it's not a sales pitch. be concise, and factual.

dury10 commented:
I have corrected the orientation and the yml file  the step file i have made an achieve and it cannot be less than 4mb @MapleLeafMakers any suggestions?

MapleLeafMakers commented:
> I have corrected the orientation and the yml file. Regarding the step file I have made an achieve and it cannot be less than 4mb @MapleLeafMakers any suggestions?

4MB is no problem, you can remove the 7z version though, just the .stp is fine

dury10 commented:
> > I have corrected the orientation and the yml file. Regarding the step file I have made an achieve and it cannot be less than 4mb @MapleLeafMakers any suggestions?
> 
> 4MB is no problem, you can remove the 7z version though, just the .stp is fine

Ok i have removed the 7z file

dury10 commented:
![image](https://github.com/VoronDesign/VoronUsers/assets/35086459/2d3745f0-1664-4485-8d43-4b847e4840ea)
@MapleLeafMakers did I miss enything?

jontek2 reviewed (commented):
Hey! Just did a quick review of your PR and here is some feedback:

1) Please update the description in the metadata.yml. Try to describe your mod with 1-2 short sentences.

2) There are some parts that requires support to be printed nicely, such as "Klicky_holder.stl" and "idler.stl". For those parts we typically model in built in supports.

3) On v6_heatsink_mod.md you have spelling errors, see "Congratuletion now let's asamble the toolhead".

Could you please take a look into these points? Thanks!

FHeilmann commented:
@dury10 Thanks for submitting this mod. After some internal discussion I have decided to reject this mod. Please find my explanation below:

Spartacus is a very complex mod to the VoronDesign 3D printer ecosystem. A lot of configurations and different variants are supported. As a result, this will likely cause the community to have a lot of feature requests and also may find some bugs while using it. You may wish to incorporate these features or fix these bugs. VoronUsers is not equipped to handle such development as feature requests and bugfixes (beyond simple things) are out of scope here. 

Here is my suggestion: **Put Spartacus into its own github repository**. This has the following advantages:

- You can structure the files as you wish, without having to adhere to our rules (which were made for mods a lot smaller)
- You can provide detailed assembly instructions and manuals which are hard to deal with in our current metadata process
- In a private repo, you are free to add your own paypal links to receive donations from the community
- The community can request features and report bugs more easily.

I'm sorry it has taken me this long to come to this conclusion, especially since you have already put a lot of work in to fix the concerns the other reviewers (and our CI) have had with the mod.

I look forward to seing Spartacus get the exposure it deserves, and the possibility of it becoming an established alternative for toolheads in the Voron printer ecosystem.

If you have any further questions, please reach out to me on discord (Haribro)

Sorry again, and best of luck

Florian (Haribro)
